pkg/kubeutils/kubeutils.go

```diff
@@ -210,6 +220,31 @@ func writeStatusMessage(writer *spinner.MessageWriter, install embeddedclusterv1
 	}
 }
 
+func WaitForHAMigration(ctx context.Context, cli client.Client) error {
+	for {
+		lastInstall, err := GetLatestInstallation(ctx, cli)
+		if err != nil {
+			return fmt.Errorf("unable to get latest installation: %v", err)
+		}
+		migrationStatus := CheckConditionStatus(lastInstall.Status, registry.RegistryMigrationStatusConditionType)
+		if migrationStatus == metav1.ConditionTrue {
+			return nil
+		}
+
+		time.Sleep(5 * time.Second)
```

Comment:
Can you change this to use a select statement instead?
One of the select conditions would be the context closing

Reply:
updated this in https://github.com/replicatedhq/embedded-cluster/pull/657/commits/451d58cf08ad79d08537578eb1b045e40c2e5c95